doc/amazon_sagemaker_debugger.rst

```diff
@@ -0,0 +1,377 @@
+
+Setup
+=====
+
+To get started, you must satisfy the following prerequisites:
+
+* Specify an AWS Region where you'll train your model.
+* To give Amazon SageMaker the access to your data in Amazon Simple Storage Service (Amazon S3) needed to train your model, create an IAM role ARN. See the `AWS IAM documentation <https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html>`__ for how to fine tune the permissions needed.
+
+Capture real-time debugging data during model training in Amazon SageMaker
+==========================================================================
+
+To enable data emission for debugging model training, Amazon SageMaker initializes a "hook" which attaches itself to the training process and emits data necessary for debugging i.e. tensors. To provide the hook's configuration, specify the new option called ``DebuggerHookConfig`` when training your model. For more about the ``DebuggerHookConfig`` object, see the `API documentation <https://sagemaker.readthedocs.io/en/stable/debugger.html#sagemaker.debugger.DebuggerHookConfig>`__.
+
+The ``DebuggerHookConfig`` accepts one or more objects of type ``CollectionConfig`` which defines the configuration around a tensor collection you intend to emit and save during model training. A "collection" helps group tensors for easier handling.
+
+.. code:: python
+
+    from sagemaker.debugger import CollectionConfig, DebuggerHookConfig
+
+    collection_config = CollectionConfig(
+        name='collection_name',
+        parameters={
+            'key': 'value'
+        }
+    )
+
+    debugger_hook_config = DebuggerHookConfig(
+        s3_output_path='s3://path/for/data/emission',
+        container_local_output_path='/local/path/for/data/emission',
+        hook_parameters={
+            'key': 'value'
+        },
+        collection_configs=[
+            collection_config
+        ]
+    )
+
+    estimator = TensorFlow(
+        role=role,
+        train_instance_count=1,
+        train_instance_type=train_instance_type,
+        debugger_hook_config=debugger_hook_config
+    )
+
+``hook_parameters`` in ``DebuggerHookConfig`` and ``parameters`` in ``CollectionConfiguration`` provide a way of fine tuning the configuration of the tensors stored during the training - they control the tensors that are saved and the step frequency at which they're saved.
+
+The possible values of ``hook_parameters`` in ``DebuggerHookConfig`` or ``parameters`` in ``CollectionConfig`` can be viewed at `CollectionParameters <https://docs.aws.amazon.com/sagemaker/latest/dg/API_CollectionConfiguration.html#SageMaker-Type-CollectionConfiguration-CollectionParameters>`__
+
+Specifying individual configuration for each collection
```

Comment:
does this need to be a separate section from the following one? it seems like they should be combined into one section as the subsections here represent steps to take

Reply:
Combined into a single section.